src/main/java/com/alipay/sofa/common/config/DefaultConfigManger.java

```diff
@@ -17,65 +17,116 @@
 package com.alipay.sofa.common.config;
 
 import com.alipay.sofa.common.utils.OrderComparator;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.util.concurrent.UncheckedExecutionException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.time.Duration;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
 
 /**
  * @author zhaowang
  * @version : SofaCommonConfig.java, v 0.1 2020年10月20日 8:30 下午 zhaowang Exp $
  */
-public class DefaultConfigManger implements ConfigManager{
+public class DefaultConfigManger implements ConfigManager {
 
     public static final Logger LOGGER = LoggerFactory.getLogger(DefaultConfigManger.class);
 
+    private final Object EMPTY = new Object();
     private final List<ConfigSource> configSources = new CopyOnWriteArrayList<>();
     private final List<ManagementListener> configListeners = new CopyOnWriteArrayList<>();
+    private LoadingCache<ConfigKey, Object> cache;
+
+    public DefaultConfigManger(long expireAfterSecond, long maximumSize) {
+        this.cache = CacheBuilder.newBuilder()
+                .expireAfterWrite(Duration.ofSeconds(expireAfterSecond))
```

Comment:
Considering call performance, refreshAfterWrites is better than expireAfterWrite.

Reply:
RefreshAfterWrites will not refresh automatically. It refreshes at the first time when `get`  method is invoked after refresh time expired.